frontend/package.json

```diff
@@ -32,6 +32,7 @@
     "bnc-notify": "^1.5.1",
     "core-js": "^3.6.5",
     "ethers": "5.7.2",
```

Comment:
*Arbitrary comment location*

One additional thing I've noticed is that the send history table does not give any indication whether the send was to the key in the stealth registry, or to the public key recovered from a transaction using advanced mode.

This got me thinking more broadly: should the public key(s) used for the send be part of the data stored in send history? And should they be displayed somehow, perhaps as a tooltip on hover or something similar. Right now I'm leaning toward "yes" as the answer to both of these questions, but I'm curious to hear what you think @alexkeating and @mds1.

Reply:
Right now we store a boolean value indicating whether the send was in an advanced mode and whether it uses a public key. Currently we don't display this in the table, but can definitely add it in this pr.

How many bits would the public key be? Asking because I think that will affect our encryption scheme. @mds1 would know better

Reply:
Compressed pubkey is 33 bytes, but yea @apbendi we're encrypting with xor and generating a unique key for each entry. So the length of the data we can encrypt is limited to the length of the key which is 32 bytes. I think just storing those two bools + the address makes sense, since you can figure out what the pubkey was based on that. If we wanted to store the pubkeys we'd have to either use PBKDF2 or whatever the current recommended key stretching algo is to stretch the viewing key, or use AES, both of which would be nice to avoid

Reply:
Prior discussion in https://github.com/ScopeLift/umbra-protocol/pull/416#discussion_r1036534952

Reply:
Per standup yesterday, let's use the remaining bytes to store as much of the pubkey as we can. To store as much of the pubkey as possible, let's reduce the two bools from bytes to half-bytes (we could use as little as one bit each, but it'll make things a bit more complex and I'd rather lean towards keeping the code simple). So what we can do is something like this:


```typescript
const buildAccountDataForEncryption = ({ address, pubKey, advancedMode, usePublicKeyChecked }: AccountDataToEncrypt) => {
  if (!isAddress(address)) {
    throw new Error('Invalid address');
  }
  address = getAddress(address).slice(2); // slice off the `0x` prefix.
  const advancedModeHalfByte = advancedMode ? '1' : '0';
  const usePublicKeyCheckedHalfByte = usePublicKeyChecked ? '1' : '0';
  const remainingBytes = 32 - 20 - 1; // 20 bytes for address, 1 byte total for advancedMode and usePublicKeyChecked.

  // Store as much of the pubkey as we can in the remaining bytes. The first 4 characters are `0x04` so we skip those.
  const pubKeyStart = pubKey.slice(4, remainingBytes + 4);

  // Concatenate all the data together.
  const dataToEncrypt = `${address}${advancedModeHalfByte}${usePublicKeyCheckedHalfByte}${pubKeyStart}`;
  if (dataToEncrypt.length !== 64) {
    // This should never happen.
    throw new Error('Data to encrypt is not the correct length');
  }

  return BigNumber.from(`0x${dataToEncrypt}`);
};
```

Reply:
Also make sure to verify that the pubkey being passed in does in fact start with `0x04` before slicing like that. (`04` is the prefix used to indicate you have an uncompressed, whereas `02` and `03` are prefixes meaning you have a compressed key (just the x-coordiante), with 2 meaning the y-coordinate is the even value and 3 meaning the y-coordinate is the odd value (there's two solutions when you solve y^2 = x^3 + 7 to uncompress))

Reply:
Agree with what @mds1 has written about including as much of the pubkey as we can. To be clear, this would be for debugging purposes. As such, I think we should use our console debugging to log the pubkeys in the console.

Going back to the advanced mode/pub key recovery discussion, I think that in the send history, next to the send address, we can include 1 or 2 emojis:

1. Sent with advanced mode on, show as:  bendi.eth <sup>🧙</sup>
2. Sent with advanced mode + pubkey checkbox enabled, show as: bendi.eth <sup>🧙🔑</sup>

Each emoji can have tooltip hover text:

🧙: "Sent with advanced mode enabled"
🔑: "Sent using recipient's standard public key"

Reply:
Added emojis https://github.com/ScopeLift/umbra-protocol/pull/494/commits/41d9b67260b60b69d78804d83e5d8dbbbed0d31e

Reply:
Hey @alexkeating, looking and working great! One thing I noticed was the copy was slightly different than above. Was there a specific reason? If not I'd prefer to match the copy as listed above. Thanks!

Reply:
Also @alexkeating I just tested with a new send and I'm not seeing the pubkey fragment being logged at all. Is that still in work?

Reply:
Sorry my bad, I misread the requirements in the above comment. Those changes will be in the next round of changes.

Reply:
Updated the copy and now we log the keys on fetch https://github.com/ScopeLift/umbra-protocol/pull/494/commits/86267ae49449fd7fb3a5614233fb8bc1e1492806

Reply:
Can you use `window.logger.debug` instead of `console.log`? We use the former for logs because it includes the version number so we know which version of the app the user is on

Reply:
Ah okay will change

Reply:
https://github.com/ScopeLift/umbra-protocol/pull/494/commits/cf55badb2206604b4f6a5442a6a5ed9b96084c13

Reply:
LGTM, thanks!